src/main/java/org/smartregister/domain/Stock.java

```diff
@@ -0,0 +1,232 @@
+package org.smartregister.domain;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Stock extends BaseDataObject {
+	
+	@JsonProperty
+	private Long identifier;
+	
+	@JsonProperty
+	private String vaccine_type_id;
+	
+	@JsonProperty
+	private String transaction_type;
+	
+	@JsonProperty
+	private String providerid;
+	
+	@JsonProperty
+	private int value;
+	
+	@JsonProperty
+	private Long date_created;
+	
+	@JsonProperty
+	private String to_from;
+	
+	@JsonProperty
+	private Long date_updated;
+	
+	@JsonProperty
+	private long version;
+
+	@JsonProperty
+	private Date deliveryDate;
+
+	@JsonProperty
+	private Date accountabilityEndDate;
+
+	@JsonProperty
+	private String donor;
+
+	@JsonProperty
+	private String serialNumber;
+
+	@JsonProperty
+	private String locationId;
+
+	@JsonProperty
+	private Map<String, String> customProperties = new HashMap();
+
+	public Stock() {
+		this.version = System.currentTimeMillis();
+	}
+	
+	public Stock(Long identifier, String vaccine_type_id, String transaction_type, String providerid, int value,
+	    Long date_created, String to_from, Long date_updated, long version) {
+		this.identifier = identifier;
+		this.vaccine_type_id = vaccine_type_id;
+		this.transaction_type = transaction_type;
+		this.providerid = providerid;
+		this.value = value;
+		this.date_created = date_created;
+		this.to_from = to_from;
+		this.date_updated = date_updated;
+		this.version = version;
+	}
+
+	public Stock(Long identifier, String vaccine_type_id, String transaction_type, String providerid, int value,
+			StockObjectMetadata stockObjectMetadata, Inventory inventory) {
+		this.identifier = identifier;
+		this.vaccine_type_id = vaccine_type_id;
+		this.transaction_type = transaction_type;
+		this.providerid = providerid;
+		this.value = value;
+		this.date_created = stockObjectMetadata.getDate_created();
+		this.to_from = stockObjectMetadata.getTo_from();
+		this.date_updated = stockObjectMetadata.getDate_updated();
+		this.version = stockObjectMetadata.getVersion();
+		this.deliveryDate = inventory.getDeliveryDate();
+		this.donor = inventory.getDonor();
+		this.serialNumber = inventory.getSerialNumber();
+		this.locationId = inventory.getServicePointId();
+		// TODO : accountabilityDate to be computed later and also set customProperties
```

Comment:
Can this be implemented

Reply:
AccountabilityEndDate is calculated using this whenever a new inventory is being added : 
```
Date accountabilityEndDate = addMonthsToDate(inventory.getDeliveryDate(),
				productCatalogue.getAccountabilityPeriod());
```
Therefore, setting it in constructor won't be of much use, because both custom properties and accountabilityenddate are set later when the actual persistence of data being taken place!

Reply:
can then the TODO be removed

Reply:
Done